src/test/java/com/github/algobot76/surabaya/AnalyzerTests.java

```diff
@@ -0,0 +1,95 @@
+package com.github.algobot76.surabaya;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.github.algobot76.surabaya.util.Analyzer;
+import com.github.algobot76.surabaya.util.Project;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+
+@SpringBootTest
+public class AnalyzerTests {
+
+	static final String testProjectDir = "src/test/testProjects/";
+
+	Analyzer analyzer;
+
+	ObjectMapper mapper;
+
+	@BeforeEach
+	public void initialize() {
+		analyzer = new Analyzer();
+		mapper = new ObjectMapper();
+	}
+
+	@Test
+	void analyzeSimpleZipFileSuccess() throws JsonProcessingException {
+		Resource testFile = new FileSystemResource(testProjectDir.concat("simpleZip.zip"));
+		Project results = analyzer.analyze(testFile);
+		String resultsString = mapper.writeValueAsString(results);
+		String expectedProjectString = "{\"packages\":{\"ast\":{\"files\":[{\"imports\":[\"Main\"],\"classes\":[{\"name\":\"DEC\",\"type\":\"Class\",\"fields\":[{\"name\":\"name\",\"type\":\"String\",\"access_modifier\":\"private\"}],\"methods\":[{\"name\":\"parse\",\"parameters\":[],\"access_modifier\":\"public\",\"return_type\":\"void\"},{\"name\":\"evaluate\",\"parameters\":[],\"access_modifier\":\"public\",\"return_type\":\"Integer\"}],\"constructors\":[],\"access_modifier\":\"public\",\"line_count\":15}]}]}}}";
```

Comment:
`expectedProjectString` is very hard to read and error-prone. Please consider building the expected `Project` manually and then convert it to a string instead.

Reply:
Yeah that is ideal and I thought about doing that, but it is way too time consuming and I don't think it is worth the effort

Reply:
Scratch that, done!